tealium/autotracking/TealiumAutotrackingModule.swift

```diff
@@ -0,0 +1,220 @@
+//
+//  TealiumAutotrackingModule.swift
+//
+//  Created by Jason Koo on 12/21/16.
+//  Copyright © 2016 Apple, Inc. All rights reserved.
+//
+
+#if TEST
+    import Foundation
+#else
+    import UIKit
+#endif
+
+enum TealiumAutotrackingKey {
+    static let moduleName = "autotracking"
+    static let eventNotificationName = "com.tealium.autotracking.event"
+    static let viewNotificationName = "com.tealium.autotracking.view"
+    static let autotracked = "autotracked"
+}
+
+extension Tealium {
+    
+    public func autotracking() -> TealiumAutotracking? {
+        
+        guard let module = modulesManager.getModule(forName: TealiumAutotrackingKey.moduleName) as? TealiumAutotrackingModule else {
+            return nil
+        }
+        
+        return module.autotracking
+        
+    }
+    
+}
+
+public protocol TealiumAutotrackingDelegate : class {
+    
+    func tealiumAutotrackShouldTrack(data: [String:Any]) -> Bool
+    func tealiumAutotrackCompleted(success:Bool, info:[String:Any]?, error:Error?)
+    
+}
+
+var tealiumAssociatedObjectHandle : UInt8 = 0
+
+public class TealiumAutotracking {
+    
+    weak var delegate : TealiumAutotrackingDelegate?
+    
+    /// Add custom data to an object, to be included with an autotracked event.
+    ///
+    /// - Parameters:
+    ///   - data: [String:Any] dictionary. Values should be String or [String]
+    ///   - toObject: Object to add data for.
+    class func addCustom(data: [String:Any], toObject:NSObject) {
```

Comment:
Vetoed.  This is actually the formatting auto-generated by quick code documentation OPT+COMMAND+/ you had found and recommended a month or two back.  Unless there's an easy way to override this default behavior, going with it will be faster than manually adding these items.